lib/internal/worker/io.js

```diff
@@ -114,7 +115,7 @@ class MessageEvent extends Event {
     this[kOrigin] = `${origin}`;
     this[kLastEventId] = `${lastEventId}`;
     this[kSource] = source;
-    this[kPorts] = [...ports];
+    this[kPorts] = ArrayPrototypeSlice(ports);
```

Comment:
Per spec, this argument should accept a `sequence<MessagePort>`, the current implementation is working as intended. Reverting…
```suggestion
    this[kPorts] = [...ports];
```